util/mtt_update.py

```diff
@@ -0,0 +1,466 @@
+		tuples.append( (root, dirs, files) )
+		t = t + 1
+
+	tuples = sorted(tuples)
+
+	paths_and_files = []
+	f = 0
+
+	for tu in tuples:
+		root = tu[0]
+		dirs = tu[1]
+		files = tu[2]
+		files = sorted(files, key=str.lower)
+		for filename in files:
+			paths_and_files.append( (os.path.join(root, filename), filename) )
+			f = f + 1
+	return paths_and_files
+
+# Walks all lua files in the mod folder, collects translatable strings,
+# and writes it to a template.txt file
+# Returns a dictionary of localized strings to source file lists
+# that can be used with the strings_to_text function.
+def generate_template(folder, mod_name):
+	dOut = {}
+	paths_and_files = sorted_os_walk(folder)
+	for paf in paths_and_files:
+		fullpath_filename = paf[0]
+		filename = paf[1]
+		if fnmatch.fnmatch(filename, "*.lua"):
+			found = read_lua_file_strings(fullpath_filename)
+			if params["verbose"]:
+				print(f"{fullpath_filename}: {str(len(found))} translatable strings")
+
+			for s in found:
+				sources = dOut.get(s, set())
+				sources.add(os.path.relpath(fullpath_filename, start=folder))
+				dOut[s] = sources
+
+	if len(dOut) == 0:
+		return None
+
+	# Convert source file set to list, sort it and add comment symbols.
+	# Needed because a set is unsorted and might result in unpredictable.
+	# output orders if any source string appears in multiple files.
+	for d in dOut:
+		sources = dOut.get(d, set())
+		sources = sorted(list(sources), key=str.lower)
+		newSources = []
+		for i in sources:
+			newSources.append(f"{symbol_source_prefix} {i} {symbol_source_suffix}")
+		dOut[d] = newSources
+
+	templ_file = os.path.join(folder, "locale/template.txt")
+	write_template(templ_file, dOut, mod_name)
+	return dOut
+
+# Updates an existing .tr file, copying the old one to a ".old" file
+# if any changes have happened
+# dNew is the data used to generate the template, it has all the
+# currently-existing localized strings
+def update_tr_file(dNew, mod_name, tr_file):
+	if params["verbose"]:
+		print(f"updating {tr_file}")
+
+	tr_import = import_tr_file(tr_file)
+	dOld = tr_import[0]
+	textOld = tr_import[1]
+
+	textNew = strings_to_text(dNew, dOld, mod_name, tr_import[2], tr_import[3])
+
+	if textOld and textOld != textNew:
+		print(f"{tr_file} has changed.")
+		if params["old-file"]:
+			shutil.copyfile(tr_file, f"{tr_file}.old")
+
+	with open(tr_file, "w", encoding='utf-8') as new_tr_file:
+		new_tr_file.write(textNew)
+
+# Updates translation files for the mod in the given folder
+def update_mod(folder):
+	modname = get_modname(folder)
+	if modname is not None:
+		print(f"Updating translations for {modname}")
+		data = generate_template(folder, modname)
+		if data == None:
+			print(f"No translatable strings found in {modname}")
+		else:
+			for tr_file in get_existing_tr_files(folder):
+				update_tr_file(data, modname, os.path.join(folder, "locale/", tr_file))
+	else:
+		print(f"\033[31mUnable to find modname in folder {folder}.\033[0m", file=_stderr)
```

Comment:
1. `cmd.exe` does not support such formatting escapes
2. Suggestion to make the message more helpful: `"Unable to determine the mod name in folder {folder}. Missing 'name' field in 'mod.conf'"`